threedi_modelchecker/config.py

```diff
@@ -2754,6 +2817,316 @@ def is_none_or_empty(col):
+            f"a {constants.CrossSectionShape.TABULATED_YZ.name} cross section shape"
+        ),
+    )
+    for col in veg_par_cols + [models.CrossSectionDefinition.friction_values]
+]
+CHECKS += [
+    CrossSectionVariableCorrectLengthCheck(
+        error_code=181,
+        column=col,
+        shapes=(constants.CrossSectionShape.TABULATED_YZ,),
+        filters=models.CrossSectionDefinition.height.is_not(None) & col.is_not(None),
+    )
+    for col in veg_par_cols
+]
+CHECKS += [
+    CrossSectionFloatListCheck(
+        error_code=187,
+        column=col,
+        shapes=(constants.CrossSectionShape.TABULATED_YZ,),
+    )
+    for col in veg_par_cols
+]
+CHECKS += [
+    QueryCheck(
+        error_code=182,
+        level=CheckLevel.WARNING,
+        column=col_csloc,
+        invalid=Query(models.CrossSectionDefinition)
+        .join(
+            models.CrossSectionLocation,
+            models.CrossSectionLocation.definition_id
+            == models.CrossSectionDefinition.id,
+        )
+        .filter(col_csloc.is_not(None) & col_csdef.is_not(None))
+        .filter(
+            models.CrossSectionLocation.friction_type.is_(constants.FrictionType.CHEZY)
+        ),
+        message=(
+            f"Both {col_csloc.table.name}.{col_csloc.name} and {col_csdef.table.name}.{col_csdef.name}"
+            f" defined without conveyance; {col_csloc.table.name}.{col_csloc.name} will be used"
+        ),
+    )
+    for col_csloc, col_csdef in [
+        (
+            models.CrossSectionLocation.vegetation_drag_coefficient,
+            models.CrossSectionDefinition.vegetation_drag_coefficients,
+        ),
+        (
+            models.CrossSectionLocation.vegetation_height,
+            models.CrossSectionDefinition.vegetation_heights,
+        ),
+        (
+            models.CrossSectionLocation.vegetation_stem_diameter,
+            models.CrossSectionDefinition.vegetation_stem_diameters,
+        ),
+        (
+            models.CrossSectionLocation.vegetation_stem_density,
+            models.CrossSectionDefinition.vegetation_stem_densities,
+        ),
+    ]
+]
+CHECKS += [
+    QueryCheck(
+        error_code=182,
+        level=CheckLevel.WARNING,
+        column=col_csloc,
+        invalid=Query(models.CrossSectionDefinition)
+        .join(
+            models.CrossSectionLocation,
+            models.CrossSectionLocation.definition_id
+            == models.CrossSectionDefinition.id,
+        )
+        .filter(col_csloc.is_not(None) & col_csdef.is_not(None))
+        .filter(
+            models.CrossSectionLocation.friction_type.is_(
+                constants.FrictionType.CHEZY_CONVEYANCE
+            )
+        ),
+        message=(
+            f"Both {col_csloc.table.name}.{col_csloc.name} and {col_csdef.table.name}.{col_csdef.name}"
+            f" defined without conveyance; {col_csdef.table.name}.{col_csdef.name} will be used"
+        ),
+    )
+    for col_csloc, col_csdef in [
```

Comment:
col_cross_section_location and col_cross_section_definition might be clearer, and consistent with the rest of the codebase

Reply:
Suggestion applied in 0fe8960f36e7ad3fb5f1437269dda3dba380279d